src/Message/DirectAuthorizeRequest.php

```diff
@@ -54,6 +54,8 @@ protected function getBaseAuthorizeData()
         $data['DeliveryPhone'] = $card->getShippingPhone();
         $data['CustomerEMail'] = $card->getEmail();
 
+        $data = array_merge($data, $this->getItemData());
```

Comment:
Hello Jason,
You are right it is much cleaner if we return the xml and assign the basketXML to the getItem Data() function.
if items array is empty and we send empty array to the basket sagepay throws error of not valid xml